apps/extension/src/ui/apps/dashboard/routes/Settings/Accounts/NewFolderModal.tsx

```diff
@@ -0,0 +1,170 @@
+
+export const NewFolderModal = () => {
+  const { t } = useTranslation("admin")
+  const { close, isOpen } = useNewFolderModal()
+
+  return (
+    <Modal open={isOpen}>
+      <ModalDialog title={t("New Folder")} onClose={close}>
+        <NewFolder onConfirm={close} onCancel={close} />
+      </ModalDialog>
+    </Modal>
+  )
+}
+
+const StyledDialog = styled(Dialog)`
+  .error {
+    font-size: var(--font-size-small);
+    color: var(--color-status-warning);
+    height: 1.6em;
+    margin-bottom: -1.6em;
+  }
+`
+
+type FormData = {
+  name: string
+  followedOnly?: boolean
+}
+
+interface NewFolderProps {
+  onConfirm: () => void
+  onCancel: () => void
+  className?: string
+}
+
+const NewFolder = ({ onConfirm, onCancel, className }: NewFolderProps) => {
+  const { t } = useTranslation("admin")
+
+  const schema = useMemo(
+    () =>
+      yup
+        .object({
+          name: yup.string().required(""),
+          followedOnly: yup.boolean(),
+        })
+        .required(),
+    []
+  )
+
+  const defaultValues = useMemo(
+    () => ({
+      name: "",
+      followedOnly: false,
+    }),
+    []
+  )
+
+  const {
+    register,
+    handleSubmit,
+    setError,
+    formState: { errors, isValid, isSubmitting },
+  } = useForm<FormData>({
+    mode: "onChange",
+    defaultValues,
+    resolver: yupResolver(schema),
+  })
+
+  const submit = useCallback(
+    async ({ name, followedOnly }: FormData) => {
+      try {
+        await api.accountsCatalogMutate([
+          { type: "addFolder", tree: followedOnly ? "watched" : "portfolio", name },
+        ])
+        onConfirm()
+      } catch (err) {
+        setError("name", {
+          type: "validate",
+          message: (err as Error)?.message ?? "",
+        })
+      }
+    },
+    [onConfirm, setError]
+  )
+
+  // "manual" field registration so we can hook our own ref to it
+  const { ref: refName, ...registerName } = register("name")
+
+  // on mount, auto select the input's text
+  const refNameRef = useRef<HTMLInputElement | null>(null)
+  useEffect(() => {
+    const input = refNameRef.current as HTMLInputElement
+    if (input) {
+      input.select()
+      input.focus()
+    }
+  }, [])
+
+  // plug both refs to the input component
+  const handleNameRef: RefCallback<HTMLInputElement> = useCallback(
+    (e: HTMLInputElement | null) => {
+      refName(e)
+      refNameRef.current = e
+    },
+    [refName]
+  )
+
+  const catalog = useAccountsCatalog()
+
+  return (
+    <StyledDialog
+      className={className}
+      text={t("Folder name")}
+      extra={
+        <form onSubmit={handleSubmit(submit)}>
+          <FormFieldContainer error={errors.name?.message}>
+            <FormFieldInputText
+              {...registerName}
+              ref={handleNameRef}
+              placeholder={t("Choose a name")}
+              spellCheck={false}
+              autoComplete="off"
+              // eslint-disable-next-line jsx-a11y/no-autofocus
+              autoFocus
```

Comment:
autoFocus in forms that are embedded in modals "break" the modal appear animation (might be why there's an eslint warning).
Best way to do it is : 
- remove autoFocus attribute
- have a useEffect with a setFocus("name") call

ex: https://github.com/TalismanSociety/talisman/blob/82d831de743ba46d31fa2572831fb605b7864c2e/apps/extension/src/ui/domains/Account/PasswordUnlock.tsx#L84